tree/dataframe/test/dataframe_snapshot.cxx

```diff
@@ -993,23 +993,28 @@ TEST(RDFSnapshotMore, ColsWithCustomTitlesMT)
 
 TEST(RDFSnapshotMore, TreeWithFriendsMT)
 {
-   const auto fname = "treewithfriendsmt.root";
-   RDataFrame(10).Define("x", []() { return 0; }).Snapshot<int>("t", fname, {"x"});
+   const auto fname1 = "treewithfriendsmt1.root";
+   const auto fname2 = "treewithfriendsmt2.root";
+   RDataFrame(10).Define("x", []() { return 42; }).Snapshot<int>("t", fname1, {"x"});
+   RDataFrame(10).Define("x", []() { return 0; }).Snapshot<int>("t", fname2, {"x"});
 
    ROOT::EnableImplicitMT();
 
-   TFile file(fname);
+   TFile file(fname1);
    auto tree = file.Get<TTree>("t");
-   TFile file2(fname);
+   TFile file2(fname2);
    auto tree2 = file2.Get<TTree>("t");
    tree->AddFriend(tree2);
 
    const auto outfname = "out_treewithfriendsmt.root";
    RDataFrame df(*tree);
-   df.Snapshot<int>("t", outfname, {"x"});
-   ROOT::DisableImplicitMT();
+   auto df_out = df.Snapshot<int>("t", outfname, {"x"});
+   EXPECT_EQ(df_out->Max<int>("x").GetValue(), 42);
+   EXPECT_EQ(df_out->GetColumnNames(), std::vector<std::string>{"x"});
```

Comment:
I was a bit confused at first because the tree with friend only shows the entries from the "x" column of the main one, instead RDF can disambiguate the friend column by prepending the name of the friend tree
```
root [0] const auto fname1 = "treewithfriendsmt1.root";
root [1] const auto fname2 = "treewithfriendsmt2.root";
root [2] TFile file(fname1);
root [3] auto tree = file.Get<TTree>("t");
root [4] TFile file2(fname2);
root [5] auto tree2 = file2.Get<TTree>("t");
root [6] tree->AddFriend(tree2);
root [9] tree->Scan("*")
************************
*    Row   *       x.x *
************************
*        0 *        42 *
*        1 *        42 *
*        2 *        42 *
*        3 *        42 *
*        4 *        42 *
*        5 *        42 *
*        6 *        42 *
*        7 *        42 *
*        8 *        42 *
*        9 *        42 *
************************
root [10] ROOT::RDataFrame df(*tree);
root [11] df.Display()->Print()
x  | t.x | 
42 | 0   | 
42 | 0   | 
42 | 0   | 
42 | 0   | 
42 | 0   | 
```
So effectively this is testing that the Snapshot can correctly select the main "x" column and not the befriended "t.x" one, right?

Reply:
that's correct!